modules/post/windows/manage/run_as_psh.rb

```diff
@@ -0,0 +1,90 @@
+##
+    # This module requires Metasploit: http://metasploit.com/download
+    # Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/post/windows/powershell'
+
+class MetasploitModule < Msf::Post
+    include Msf::Post::Windows::Powershell
+    def initialize(info={})
+        super( update_info( info,
+            'Name'          => 'Windows Shell As Another User',
+            'Description'   => %q{ This module will start a process as another user using powershell. },
+            'License'       => MSF_LICENSE,
+            'Author'        => [ 'p3nt4' ],
+            'Platform'      => [ 'win' ],
+            'SessionTypes'  => [ 'meterpreter' ]
+        ))
+        register_options(
+            [
+                OptString.new('USER', [true, 'User to run executable as', nil]),
+                OptString.new('PASS', [true, 'Password of user', nil]),
+                OptString.new('DOMAIN', [false, 'Domain of user', '']),
+                OptString.new('EXE', [true, 'Executable to run', 'cmd.exe']),
+                OptString.new('ARGS', [false, 'Arguments', nil]),
+                OptString.new('PATH', [true, 'Working Directory', 'C:\\']),
+                OptBool.new('CHANNELIZE', [true, 'Chanelize output, required for reading output or interracting', true]),
+                OptBool.new('INTERACTIVE', [true, 'Run interactively', true]),
+                OptBool.new('HIDDEN', [true, 'Hide the window', true]),
+                #OptString.new('DUMMY', [false, 'Run in a dummy host process', '']),
+            ], self.class)
+    end
+
+    def run
+        begin
```

Comment:
Only the relevant section where there is a chance of error should be wrapped in the begin.